odoo/addons/base/models/res_partner.py

```diff
@@ -239,6 +239,25 @@ def default_get(self, default_fields):
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name'),
     ]
 
+    def _compute_avatar_1920(self):
+        return self._compute_avatar_xxxx('1920')
+
+    def _compute_avatar_128(self):
+        return self._compute_avatar_xxxx('128')
+
+    def _compute_avatar_xxxx(self, field):
+        internal_users = self.user_ids - self.user_ids.filtered('share')
+        for record in self:
+            if record['image_%s' % field]:
+                record['avatar_%s' % field] = record['image_%s' % field]
+            else:
+                record['avatar_%s' % field] = record.generate_avatar_svg(record.name) if record.user_ids in internal_users else record._get_placeholder()
```

Comment:
It makes little functional sense to test for `internal_users` on a company.
I would be expecting the conditions to be split into: is company else, is internal, is partner